src/apps/content/views.py

```diff
@@ -1,9 +1,40 @@
+import json
+
+from django.db.models import Case, When
 from django.views.generic import TemplateView
 
+from src.apps.inventory.models import Product
+from src.apps.inventory.redis_utils import RedisConnection
+
 
 class IndexView(TemplateView):
     template_name = "content/index.html"
 
+    def get_product_views(self):
+        redis_conn = RedisConnection()
+        redis_connection = redis_conn.get_connection()
+
+        product_views_data = redis_connection.get("product:views")
+
+        if product_views_data:
+            product_views = json.loads(product_views_data)
```

Comment:
Well, 

No, definitely. no

It means you store everything as plain text and read all data from key on each request? You are gonna blow our little ecom.

Take a look at redis data structures https://redis.com/redis-enterprise/data-structures/

I think we can use SORTED SET.
By default SET will store only unique values, use ProductID for that. And SORTED SETS assign weight to every record. We don't need to read all records then, we can just ask SORTED SET for top N items and that is how we are gonna get our most viewed products.

Lets name our data structure "most_viewed_products" and save it as constant value somewhere, so it will be easier for us to generate keys in the future.